File: src/main/java/io/github/acopier/raildest/utilities/DestinationData.java
package io.github.acopier.raildest.utilities;

import io.github.acopier.raildest.RailDestPlugin;
import org.bukkit.NamespacedKey;
import org.bukkit.entity.Player;
import org.bukkit.persistence.PersistentDataType;

public class DestinationData {

  private static final NamespacedKey KEY =
      new NamespacedKey(RailDestPlugin.PLUGIN_ID, "destination");

  public static void setDestination(Player player, String destination) {
    if (destination == null || destination.isEmpty()) {
      player.getPersistentDataContainer().remove(KEY);
      return;
    }
    player.getPersistentDataContainer().set(KEY, PersistentDataType.STRING, destination);
  }

  public static String getDestination(Player player) {
    return player.getPersistentDataContainer().getOrDefault(KEY, PersistentDataType.STRING, "");
  }
}

File: src/main/kotlin/io/github/acopier/raildest/utilities/DestinationData.kt
package io.github.acopier.raildest.utilities

import io.github.acopier.raildest.RailDest
import org.bukkit.NamespacedKey
import org.bukkit.entity.Player
import org.bukkit.persistence.PersistentDataType


object DestinationData {
    private val KEY = RailDest.plugin?.let {
        NamespacedKey(
            it, "destination"
        )
    }

    fun setDestination(player: Player, destination: String?) {
        if (destination == null || destination.isEmpty()) {
            KEY?.let { player.persistentDataContainer.remove(it) }
            return
        }
        KEY?.let {
            player.persistentDataContainer.set(
                it,
                PersistentDataType.STRING,
                destination
            )
        }
    }

    fun getDestination(player: Player): String? {
        return KEY?.let {
            player.persistentDataContainer.getOrDefault(
                it,
                PersistentDataType.STRING,
                ""
            )
        }
    }
}
